### PR TITLE
Removing outdated limitations in Searchable Snapshots describing KNN is not compatible.

### PR DESCRIPTION
**Pull request by 0ctopus13prime**
### Description
Since 2.18, now snapshot supports KNN index. Hence removing the comments saying KNN index is not supported.

### Issues Resolved
https://github.com/opensearch-project/k-NN/issues/2033

### Version
2.18

### Frontend features
N/A

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _tuning-your-cluster/availability-and-recovery/snapshots/searchable_snapshot.md [hunk 1]
```diff
@@ -109,4 +109,3 @@ The following are known limitations of the searchable snapshots feature:
 - Searching remote data can impact the performance of other queries running on the same node. We recommend that users provision dedicated nodes with the `search` role for performance-critical applications.
 - For better search performance, consider [force merging]({{site.url}}{{site.baseurl}}/api-reference/index-apis/force-merge/) indexes into a smaller number of segments before taking a snapshot. For the best performance, at the cost of using compute resources prior to snapshotting, force merge your index into one segment.
 - We recommend configuring a maximum ratio of remote data to local disk cache size using the `cluster.filecache.remote_data_ratio` setting. A ratio of 5 is a good starting point for most workloads to ensure good query performance. If the ratio is too large, then there may not be sufficient disk space to handle the search workload. For more details on the maximum ratio of remote data, see issue [#11676](https://github.com/opensearch-project/OpenSearch/issues/11676).
-- k-NN native-engine-based indexes using `faiss` and `nmslib` engines are incompatible with searchable snapshots.
```
Comment (navneet1v) [inline _tuning-your-cluster/availability-and-recovery/snapshots/searchable_snapshot.md:112]: I know we are removing this line here, but somewhere I was thinking we should add that this feature is supported with k-NN indices. May be before `Create a searchable snapshot index` we can add a note saying from 2.18 version of Opensearch Searchable snapshots are supported with K-NN indices.

Comment (0ctopus13prime) [reply _tuning-your-cluster/availability-and-recovery/snapshots/searchable_snapshot.md:112]: Updated.

File: _tuning-your-cluster/availability-and-recovery/snapshots/searchable_snapshot.md [hunk 2]
```diff
@@ -109,4 +109,4 @@ The following are known limitations of the searchable snapshots feature:
...
+- Since 2.18, k-NN indices (NMSLIB and FAISS) now support searchable snapshots.
```
Comment (navneet1v) [inline _tuning-your-cluster/availability-and-recovery/snapshots/searchable_snapshot.md:112]: this is limitation section.

Comment (0ctopus13prime) [reply _tuning-your-cluster/availability-and-recovery/snapshots/searchable_snapshot.md]: oh shoot, my bad.
I just move the line before the section you mentioned.

**Comment by kolchfa-aws** [comment]
@0ctopus13prime Is this PR ready for doc review?

**Comment by navneet1v** [comment]
> @0ctopus13prime Is this PR ready for doc review?

yes it is

File: _tuning-your-cluster/availability-and-recovery/snapshots/searchable_snapshot.md [hunk 3]
```diff
@@ -46,7 +46,7 @@ services:
       - node.search.cache.size=50gb
 ```
 
-
+- Since 2.18, k-NN indices (NMSLIB and FAISS) now support searchable snapshots.
```
Comment (kolchfa-aws) [inline _tuning-your-cluster/availability-and-recovery/snapshots/searchable_snapshot.md:49]:
```suggestion
- Starting with version 2.18, k-NN indexes now support searchable snapshots for the NMSLIB and Faiss engines.
```

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @0ctopus13prime! One suggestion.

**Comment by 0ctopus13prime** [comment]
@kolchfa-aws 
Updated accordingly! Thank you for the suggestion

**Review by natebower — APPROVED** [review]
@0ctopus13prime @kolchfa-aws LGTM!

**Pull request merged**